### PR TITLE
Refresh gradient clip patch.

### PR DESCRIPTION
**Pull request by dlibenzi**

### CODE REVIEW

File: torch_patches/X10-clip_grad.diff [hunk 1]
```diff
@@ -20,7 +22,7 @@ index d088c4b37..685182206 100644
 +            param_norm = p.grad.data.norm(norm_type) ** norm_type
 +            total_norm.add_(param_norm)
 +        total_norm = (total_norm ** (1. / norm_type))
-+    clip_coef = torch.tensor(max_norm, device=device) / (total_norm + 1e-6)
++        clip_coef = torch.tensor(max_norm, device=device) / (total_norm + 1e-6)
 +    for p in parameters:
 +        p.grad.data.mul_(torch.where(clip_coef < 1, clip_coef, torch.tensor(1., device=device)))
```
Comment (pietern) [inline torch_patches/X10-clip_grad.diff]: `device` may still be undefined.

Comment (dlibenzi) [reply torch_patches/X10-clip_grad.diff]: > `device` may still be undefined.

Duh! Of course.
One more try ...

**Comment by pietern** [comment]
I can't tell if this is still equivalent from a quick glance, but otherwise LGTM.

Deferring to someone else for approval.

**Comment by pietern** [comment]
Thanks for the quick patch, @dlibenzi!
